### PR TITLE
feat: [SSCA-1175]: Add RBAC in Remediation APIs

### PR DESCRIPTION
**Pull request by Varsha-lalwani**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/57158)

### CODE REVIEW

**Comment by Varsha-lalwani** [comment]
trigger uts

File: ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java [hunk 1]
```diff
@@ -44,46 +53,55 @@ public class RemediationTrackerApiImpl implements RemediationApi {
...
     RemediationTrackerCreateResponseBody response = new RemediationTrackerCreateResponseBody().id(remediationTrackerId);
     return Response.ok().entity(response).build();
   }
 
   @Override
-  public Response updateRemediationTracker(String org, String project, String remediation,
-      @Valid RemediationTrackerUpdateRequestBody body, String harnessAccount) {
+  @NGAccessControlCheck(resourceType = REMEDIATION_TRACKER_RESOURCE, permission = REMEDIATION_TRACKER_EDIT)
+  public Response updateRemediationTracker(@OrgIdentifier String org, @ProjectIdentifier String project,
+      String remediation, @Valid RemediationTrackerUpdateRequestBody body, @AccountIdentifier String harnessAccount) {
```
Comment (reetika04) [inline ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java:68]: Add ResourceIdentifier annotation for remediation

File: ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java [hunk 2]
```diff
@@ -44,46 +53,55 @@ public class RemediationTrackerApiImpl implements RemediationApi {
...
         remediationTrackerService.updateRemediationTracker(harnessAccount, org, project, remediation, body);
     RemediationTrackerUpdateResponseBody response = new RemediationTrackerUpdateResponseBody().id(remediationTrackerId);
     return Response.ok().entity(response).build();
   }
 
   @Override
-  public Response closeRemediationTracker(String org, String project, String remediation, String harnessAccount) {
+  @NGAccessControlCheck(resourceType = REMEDIATION_TRACKER_RESOURCE, permission = REMEDIATION_TRACKER_CLOSE)
+  public Response closeRemediationTracker(@OrgIdentifier String org, @ProjectIdentifier String project,
+      String remediation, @AccountIdentifier String harnessAccount) {
```
Comment (reetika04) [inline ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java:78]: Add ResourceIdentifier annotation for remediation

File: ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java [hunk 3]
```diff
@@ -44,46 +53,55 @@ public class RemediationTrackerApiImpl implements RemediationApi {
...
     boolean response = remediationTrackerService.close(harnessAccount, org, project, remediation);
     return Response.ok().entity(new SaveResponse().status(response ? "SUCCESS" : "FAILURE")).build();
   }
 
   @Override
-  public Response excludeArtifact(
-      String org, String project, String remediation, @Valid ExcludeArtifactRequest body, String harnessAccount) {
+  @NGAccessControlCheck(resourceType = REMEDIATION_TRACKER_RESOURCE, permission = REMEDIATION_TRACKER_EDIT)
+  public Response excludeArtifact(@OrgIdentifier String org, @ProjectIdentifier String project, String remediation,
```
Comment (reetika04) [inline ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java:85]: Same

File: ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java [hunk 4]
```diff
@@ -92,33 +110,39 @@ public Response listRemediations(String org, String project, @Valid RemediationL
   }
 
   @Override
-  public Response getRemediationDetails(String org, String project, String remediation, String harnessAccount) {
+  @NGAccessControlCheck(resourceType = REMEDIATION_TRACKER_RESOURCE, permission = REMEDIATION_TRACKER_VIEW)
+  public Response getRemediationDetails(@OrgIdentifier String org, @ProjectIdentifier String project,
+      String remediation, @AccountIdentifier String harnessAccount) {
```
Comment (reetika04) [inline ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java:115]: here as well

File: ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java [hunk 5]
```diff
@@ -128,18 +152,20 @@ public Response getDeploymentsListForArtifactInRemediation(String org, String pr
   }
 
   @Override
-  public Response getEnvironmentListForRemediation(
-      String org, String project, String remediation, String artifact, String harnessAccount, String envType) {
+  @NGAccessControlCheck(resourceType = REMEDIATION_TRACKER_RESOURCE, permission = REMEDIATION_TRACKER_VIEW)
+  public Response getEnvironmentListForRemediation(@OrgIdentifier String org, @ProjectIdentifier String project,
+      String remediation, String artifact, @AccountIdentifier String harnessAccount, String envType) {
```
Comment (reetika04) [inline ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java:157]: here

File: ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java [hunk 6]
```diff
@@ -128,18 +152,20 @@ public Response getDeploymentsListForArtifactInRemediation(String org, String pr
...
         harnessAccount, org, project, remediation, artifact, env);
     return Response.ok().entity(response).build();
   }
 
   @Override
-  public Response createTicket(
-      String orgId, String projectId, String remediation, @Valid CreateTicketRequest body, String harnessAccount) {
-    String ticketId = remediationTrackerService.createTicket(projectId, remediation, orgId, body, harnessAccount);
+  @NGAccessControlCheck(resourceType = REMEDIATION_TRACKER_RESOURCE, permission = REMEDIATION_TRACKER_EDIT)
+  public Response createTicket(@ProjectIdentifier String project, String remediation, @OrgIdentifier String org,
```
Comment (reetika04) [inline ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java:166]: here

File: ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java [hunk 7]
```diff
@@ -92,33 +110,39 @@ public Response listRemediations(String org, String project, @Valid RemediationL
...
         remediationTrackerService.getRemediationArtifactDetails(harnessAccount, org, project, remediation, artifact);
     return Response.ok().entity(response).build();
   }
 
   @Override
-  public Response getDeploymentsListForArtifactInRemediation(String org, String project, String remediation,
-      String artifact, @Valid RemediationArtifactDeploymentsListingRequestBody body, String harnessAccount,
+  @NGAccessControlCheck(resourceType = REMEDIATION_TRACKER_RESOURCE, permission = REMEDIATION_TRACKER_VIEW)
+  public Response getDeploymentsListForArtifactInRemediation(@OrgIdentifier String org,
+      @ProjectIdentifier String project, String remediation, String artifact,
```
Comment (reetika04) [inline ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java:144]: here

File: ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java [hunk 8]
```diff
@@ -92,33 +110,39 @@ public Response listRemediations(String org, String project, @Valid RemediationL
...
         remediationTrackerService.listRemediationArtifacts(harnessAccount, org, project, remediation, body, pageable);
     return PageResponseUtils.getPagedResponse(artifactEntities);
   }
 
   @Override
-  public Response getArtifactInRemediationDetails(
-      String org, String project, String remediation, String artifact, String harnessAccount) {
+  @NGAccessControlCheck(resourceType = REMEDIATION_TRACKER_RESOURCE, permission = REMEDIATION_TRACKER_VIEW)
+  public Response getArtifactInRemediationDetails(@OrgIdentifier String org, @ProjectIdentifier String project,
+      String remediation, String artifact, @AccountIdentifier String harnessAccount) {
```
Comment (reetika04) [inline ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java:135]: here

File: ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java [hunk 9]
```diff
@@ -92,33 +110,39 @@ public Response listRemediations(String org, String project, @Valid RemediationL
...
     return Response.ok().entity(response).build();
   }
 
   @Override
-  public Response getArtifactListForRemediation(String org, String project, String remediation,
-      @Valid RemediationArtifactListingRequestBody body, String harnessAccount, @Min(1L) @Max(1000L) Integer limit,
-      @Min(0L) Integer page) {
+  @NGAccessControlCheck(resourceType = REMEDIATION_TRACKER_RESOURCE, permission = REMEDIATION_TRACKER_VIEW)
+  public Response getArtifactListForRemediation(@OrgIdentifier String org, @ProjectIdentifier String project,
+      String remediation, @Valid RemediationArtifactListingRequestBody body, @AccountIdentifier String harnessAccount,
```
Comment (reetika04) [inline ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java:124]: here

**Comment by Varsha-lalwani** [comment]
trigger uts

**Comment by Varsha-lalwani** [comment]
trigger uts

**Review by reetika04 — APPROVED** [review]

**Pull request merged**